### PR TITLE
Add APIs for Key Vault 4.1.0

### PR DESCRIPTION
**Pull request by heaths**

### CODE REVIEW

**Comment by heaths** [comment]
Note that checks will fail because the tests have to be re-recorded, but the 7.1 deployment is not yet complete. I also figured it would be faster to review the PR without the many .json files that will change, which will cause the browser-based PR to render and respond poorly.

File: sdk/keyvault/Azure.Security.KeyVault.Keys/api/Azure.Security.KeyVault.Keys.netstandard2.0.cs [hunk 1]
```diff
@@ -159,7 +160,9 @@ public static partial class KeyModelFactory
...
         public static Azure.Security.KeyVault.Keys.JsonWebKey JsonWebKey(Azure.Security.KeyVault.Keys.KeyType keyType, string id = null, System.Collections.Generic.IEnumerable<Azure.Security.KeyVault.Keys.KeyOperation> keyOps = null, Azure.Security.KeyVault.Keys.KeyCurveName? curveName = default(Azure.Security.KeyVault.Keys.KeyCurveName?), byte[] d = null, byte[] dp = null, byte[] dq = null, byte[] e = null, byte[] k = null, byte[] n = null, byte[] p = null, byte[] q = null, byte[] qi = null, byte[] t = null, byte[] x = null, byte[] y = null) { throw null; }
-        public static Azure.Security.KeyVault.Keys.KeyProperties KeyProperties(System.Uri id = null, System.Uri vaultUri = null, string name = null, string version = null, bool managed = false, System.DateTimeOffset? createdOn = default(System.DateTimeOffset?), System.DateTimeOffset? updatedOn = default(System.DateTimeOffset?), string recoveryLevel = null) { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
```
Comment (christothes) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/api/Azure.Security.KeyVault.Keys.netstandard2.0.cs:163]: Should this be added to the other model factory methods or is there a reason it's just on this one?

**Review by christothes — APPROVED** [review]

File: sdk/keyvault/Azure.Security.KeyVault.Keys/api/Azure.Security.KeyVault.Keys.netstandard2.0.cs (hunk 1 above)
Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Keys/api/Azure.Security.KeyVault.Keys.netstandard2.0.cs:163]: The rule for overloading factory methods (or any methods, really) is to hide the old one, make it's members non-default, and the "longest" overload (most parameters) the one with defaults. New code compiling against this version of the assembly ends up using the right one (though it really doesn't matter if they don't actually use the new parameter).

<https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-mocking-factory-builder-methods>

**Review by tg-msft — APPROVED** [review]
Public API changes look good

**Pull request merged**